spec/models/claim/matching_attribute_finder_spec.rb

```diff
@@ -0,0 +1,83 @@
+require "rails_helper"
+
+RSpec.describe Claim::MatchingAttributeFinder do
+  describe "#claim_ids_with_matching_attributes" do
+    let(:source_claim) {
+      create(
+        :claim,
+        teacher_reference_number: "0902344",
+        national_insurance_number: "QQ891011C",
+        email_address: "genghis.khan@mongol-empire.com",
+        bank_account_number: "34682151",
+        bank_sort_code: "972654",
+        building_society_roll_number: "123456789/ABCD",
+      )
+    }
+
+    let!(:claim_with_no_matching_attributes) { create(:claim, :submitted) }
+    let!(:unsubmitted_claim_with_matching_teacher_reference_number) { create(:claim, :submittable, teacher_reference_number: source_claim.teacher_reference_number) }
+
+    let(:claim_with_matching_attribute) { build(:claim, :submitted) }
+    let(:claims_with_matching_attributes) { Claim::MatchingAttributeFinder.new(source_claim).claim_ids_with_matching_attribute_keys }
+
+    it "does not include the id of the source claim" do
```

Comment:
I'd like to get @tekin's opinion on the structure of the first three `it` blocks here, each of which calls the `#claim_ids_with_matching_attribute_keys` method, making various assertions each time but with the same setup data. They could be combined into a single `it` which would speed up the tests, maybe at the cost of readability. We discussed something similar on a PR Tekin reviewed for me last week.